File: gs-messaging-jms/complete-scala/src/main/scala/hello/Application.scala

package hello

import org.springframework.boot.{CommandLineRunner, SpringApplication}
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.jms.core.JmsTemplate
import org.springframework.jms.core.MessageCreator
import org.springframework.util.FileSystemUtils
import javax.jms.{Message, Session, TextMessage}
import java.io.File

import org.springframework.beans.factory.annotation.Autowired

import scala.beans.BeanProperty

@SpringBootApplication
class Application extends CommandLineRunner {

  @BeanProperty
  @Autowired
  val jmsTemplate : JmsTemplate = null

  def run(args: String*) : Unit = {
    // Clean out any ActiveMQ data from a previous run
    FileSystemUtils.deleteRecursively(new File("activemq-data"))

    // create message
    val messageCreator : MessageCreator = new MessageCreator {
      override def createMessage(session: Session): Message = session.createTextMessage("ping!")
    }

    val receiver: Runnable = new Runnable {
      def run(): Unit ={
        // synchronous reception
        System.out.println("====> Waiting for new message.")
        jmsTemplate.setReceiveTimeout(100)
        jmsTemplate.receive("default") match {
          case m:TextMessage => println("====> Synchronous reception: " + m.getText)
          case _ => println("====> Synchronous reception: No message received")
        }
      }
    }

    new Thread(receiver).start()

    // Send a message
    System.out.println("====> Sending a new message.")
    jmsTemplate.send("default", messageCreator)
  }
}

object Application extends App {
  SpringApplication.run(classOf[Application], args:_*)
}

File: gs-messaging-jms/complete-java/src/main/java/hello/Application.java

package hello;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.jms.core.JmsTemplate;
import org.springframework.jms.core.MessageCreator;
import org.springframework.util.FileSystemUtils;

import javax.jms.Message;
import java.io.File;

@SpringBootApplication
public class Application {
    public static void main(String[] args) {
        // Clean out any ActiveMQ data from a previous run
        FileSystemUtils.deleteRecursively(new File("activemq-data"));

        // Launch the application
        ConfigurableApplicationContext context = SpringApplication.run(Application.class, args);

        JmsTemplate jmsTemplate = context.getBean(JmsTemplate.class);

        // create message
        MessageCreator messageCreator = session -> session.createTextMessage("ping!");

        final Runnable receiver = () -> {
            // synchronous reception
            System.out.println("====> Waiting for new message.");
            jmsTemplate.setReceiveTimeout(100);
            final Message message = jmsTemplate.receive("default");
            if (message == null){
                System.out.println("====> Synchronous reception: no message received");
            } else {
                System.out.println("====> Synchronous reception: " + message);
            }
        };
        new Thread(receiver).start();

        // Send a message
        System.out.println("====> Sending a new message.");
        jmsTemplate.send("default", messageCreator);
    }

}
